### PR TITLE
Alfresco 5.0.1 supports Java 8, but Maven Compiler settings are hardcode...

### PR DESCRIPTION
**Pull request by rpiccoli**
Alfresco 5.0.1 supports Java 8, but Maven Compiler settings are hardcoded to 1.7.

With this change, compiling Java 8 code for 5.0.1 is possible by overwriting the properties:

maven.compiler.target : 1.8
maven.compiler.source: 1.8

### CODE REVIEW

**Comment by mindthegab** [comment]
Ciao @rpiccoli,

I think the right resolution for this is to update the compiler / enforcer settings to allow also Java8.

Also, re your patch, are you aware that:

 maven.compiler.source
 maven.compiler.target

are actually the global 'user properties' to configure the Maven Compiler behavior (see [here](http://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#source)] so you can just set them in the <properties> section without having to explicitly filter them in the maven-compiler-plugin configuration?

Basically the patch would be OK without the line 234/235 change.

File: poms/alfresco-sdk-parent/pom.xml [hunk 1]
```diff
@@ -231,8 +233,8 @@
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>${maven.compiler.version}</version>
                     <configuration>
-                        <source>1.7</source>
-                        <target>1.7</target>
+                        <source>${maven.compiler.source}</source>
```
Comment (mindthegab) [inline poms/alfresco-sdk-parent/pom.xml]: These 2 lines are unneeded since maven.compiler.source and maven.compiler.target are the user properties so can be configured in <properties> and <properties> only.

**Comment by rpiccoli** [comment]
As requested, I removed the redundant lines.
